Firestore/Example/Tests/Integration/API/FIRCompositeIndexQueryTests.mm

```diff
@@ -0,0 +1,206 @@
+ */
+
+#import <FirebaseFirestore/FirebaseFirestore.h>
+#import "Firestore/Example/Tests/Util/FSTIntegrationTestCase.h"
+
+#include "Firestore/core/src/util/autoid.h"
+
+using firebase::firestore::util::CreateAutoId;
+
+/**
+ * This FIRCompositeIndexQueryTests class is designed to facilitate integration
+ * testing of Firestore queries that require composite indexes within a
+ * controlled testing environment.
+ *
+ * Key Features:
+ * - Runs tests against the dedicated test collection with predefined composite
+ *   indexes.
+ * - Automatically associates a test ID with documents for data isolation.
+ * - Utilizes TTL policy for automatic test data cleanup.
+ * - Constructs Firestore queries with test ID filters.
+ */
+@interface FIRCompositeIndexQueryTests : FSTIntegrationTestCase
+// Creates a new unique identifier for each test case to ensure data isolation.
+@property(nonatomic, strong) NSString *testId;
+@end
+
+static NSString *const TEST_ID_FIELD = @"testId";
+static NSString *const COMPOSITE_INDEX_TEST_COLLECTION = @"composite-index-test-collection";
+
+@implementation FIRCompositeIndexQueryTests
+
+- (void)setUp {
+  [super setUp];
+  self.testId = [NSString stringWithFormat:@"test-id-%s", CreateAutoId().c_str()];
+}
+
+#pragma mark - Test Helpers
+
+// Return reference to the static test collection: composite-index-test-collection
+- (FIRCollectionReference *)testCollectionRef {
+  return [self.db collectionWithPath:COMPOSITE_INDEX_TEST_COLLECTION];
+}
+
+// Runs a test with specified documents in the COMPOSITE_INDEX_TEST_COLLECTION.
+- (FIRCollectionReference *)withCompositeIndexTestDocs:
+    (NSDictionary<NSString *, NSDictionary<NSString *, id> *> *)docs {
+  FIRCollectionReference *writer = [self testCollectionRef];
+  // Use a different instance to write the documents
+  [self writeAllDocuments:[self prepareTestDocuments:docs]
+             toCollection:[self.firestore collectionWithPath:writer.path]];
+  return self.testCollectionRef;
+}
+
+// Helper method to hash document keys and add test-specific fields for the provided documents.
+- (NSDictionary<NSString *, NSDictionary<NSString *, id> *> *)prepareTestDocuments:
+    (NSDictionary<NSString *, NSDictionary<NSString *, id> *> *)docs {
+  NSMutableDictionary<NSString *, NSDictionary<NSString *, id> *> *result =
+      [NSMutableDictionary dictionaryWithCapacity:docs.count];
+  for (NSString *key in docs.allKeys) {
+    NSDictionary<NSString *, id> *doc = docs[key];
+    NSDictionary<NSString *, id> *updatedDoc = [self addTestSpecificFieldsToDoc:doc];
+    result[[self toHashedId:key]] = updatedDoc;
+  }
+  return [result copy];
+}
+
+// Adds test-specific fields to a document, including the testId and expiration date.
+- (NSDictionary<NSString *, id> *)addTestSpecificFieldsToDoc:(NSDictionary<NSString *, id> *)doc {
+  NSMutableDictionary<NSString *, id> *updatedDoc = [doc mutableCopy];
+  updatedDoc[TEST_ID_FIELD] = self.testId;
+  NSTimeInterval expirationTime =
```

Comment:
I think you should use `int64_t` instead of `NSTimeInterval`?

`[[FIRTimestamp timestamp] seconds]` returns an `int64_t`, and `timestampWithSeconds:` expects an `int64_t`.